### PR TITLE
[WIP] [core] [object store] Support HCCL as a tensor transport backend with object store

### PR DESCRIPTION
**Pull request by Kishanthan**
## Why are these changes needed?

Ray Collective API support for communication between NPUs using HCCL is being added with #55381. On top of it, this PR adds HCCL as the tensor transport for object store as below.

```python
import sys
import torch
import pytest
import ray
from ray.experimental.collective import create_collective_group


@ray.remote(num_cpus=0, resources={"NPU": 1}, enable_tensor_transport=True)
class NPUTestActor:
    @ray.method(tensor_transport="hccl")
    def echo(self, data):
        return data.to("npu")

    def sum(self, data):
        return data.sum().item()


@pytest.mark.parametrize(
    "ray_start_regular", [{"resources": {"NPU": 2}}], indirect=True
)
def test_hccl(ray_start_regular):
    world_size = 2
    actors = [NPUTestActor.remote() for _ in range(world_size)]
    create_collective_group(actors, backend="hccl", name="test_hccl_group")

    src_actor, dst_actor = actors[0], actors[1]

    # Create test tensor
    tensor = torch.tensor([1, 2, 3])
    gpu_ref = src_actor.echo.remote(tensor)

    # Trigger tensor transfer from src to dst actor
    remote_sum = ray.get(dst_actor.sum.remote(gpu_ref))
    assert tensor.sum().item() == remote_sum


if __name__ == "__main__":
    sys.exit(pytest.main(["-sv", __file__]))
```

Depends on #55381

## Related issue number

#55364 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a
           method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Kishanthan** [comment]
> This pull request has been automatically marked as stale because it has not had any activity for 14 days. It will be closed in another 14 days if no further activity occurs. Thank you for your contributions.
> 
> You can always ask for help on our [discussion forum](https://discuss.ray.io/) or [Ray's public slack channel](https://github.com/ray-project/ray#getting-involved).
> 
> If you'd like to keep this open, just leave any comment, and the stale label will be removed.

This is not stale. Waiting for the dependent PR to be merged.

File: python/ray/util/collective/collective_group/hccl_collective_group.py [hunk 1]
```diff
@@ -0,0 +1,819 @@
...
+                f"HCCLGroup expected metadata in internal_kv with name `{metadata_key}`. "
+                "HCCLGroup should not be instantiated directly. "
+                "Use ray.experimental.collective.create_collective_group to create the group."
+            )
+
+        metadata = metadata.decode()
+        master_addr, master_port = metadata.split(":")
+        os.environ["MASTER_ADDR"] = master_addr
+        os.environ["MASTER_PORT"] = master_port
+        dist.init_process_group(backend="hccl", rank=rank, world_size=world_size)
```
Comment (tianyi-ge) [inline python/ray/util/collective/collective_group/hccl_collective_group.py:150]: I wonder if `init_process_group` is this necessary? This function can only be called once for each process. I guess this is for `util.collective`? Can `util.collective` create a group lazily by `comm_init_task` (call the raw c `HcclCommInitRootInfo` function)?

Comment (yanggthomas) [reply python/ray/util/collective/collective_group/hccl_collective_group.py:150]: +1

Comment (tianyi-ge) [reply python/ray/util/collective/collective_group/hccl_collective_group.py:150]: I tested this PR with verl. As long as I removed this `init_process_group`, everything works fine. Otherwise an exception is raised due to duplicate communication group initialization.
